mypy/fscache.py

```diff
@@ -209,7 +209,7 @@ def isfile_case(self, path: str, prefix: str) -> bool:
 
         # Also check the other path components in case sensitive way.
         head, dir = os.path.split(head)
-        while res and head and dir and prefix in head:
+        while res and head and dir and head.startswith(prefix):
```

Comment:
This seems good enough. Perhaps you could see if adding `assert head.startswith(prefix)` before line 211 always passes? OTOH perhaps that's risking too much.

I have a faint worry that e.g. `head = '/foo/bar'` and `prefix = '/foo/b'` but that's not possible given how `prefix` is constructed by the caller right?

Reply:
> Perhaps you could see if adding `assert head.startswith(prefix)` before line 211 always passes? OTOH perhaps that's risking too much.

I added the assert, and all tests still passed. I think however it is better to not put it there, just in case.

> I have a faint worry that e.g. `head = '/foo/bar'` and `prefix = '/foo/b'` but that's not possible given how prefix is constructed by the caller right?

Yes. I will probably just clarify this in the docstring.